### PR TITLE
Integrate past usernames tooltip to user profile overlay

### LINKED ISSUE #24737: User profile overlay doesn't show previous username history
### Discussed in https://github.com/ppy/osu/discussions/24731

<div type='discussions-op-text'>

<sup>Originally posted by **soopax** September  7, 2023</sup>
Lazer doesn't show this feature. On the website on a player's profile, if they have changed their name, their previous names are listed with an icon near their current name.

### Screenshots

Website
web

Website; hovering over the icon expands it to show the player's username history
web-expanded

-------------
Lazer; there is no icon
lazer

</div>

### PR DESCRIPTION
**Pull request by ItsShamed**
Closes #24737

osu Game Tests_GsZqfBmYEx

Just a matter of tweaking size and colour to match web as close as possible.

### CODE REVIEW

File: osu.Game/Graphics/OsuColour.cs [hunk 1]
```diff
@@ -383,6 +383,7 @@ public static Color4 ForegroundTextColourFor(Color4 backgroundColour)
 
         // Content Background
         public readonly Color4 B5 = Color4Extensions.FromHex(@"222a28");
+        public readonly Color4 B6 = Color4Extensions.FromHex(@"1c1719"); // hsl(333,10%,10%)
```
Comment (bdach) [inline osu.Game/Graphics/OsuColour.cs:386]: This is not a thing that should need adding. The colour you want already exists as `Background6` in `OverlayColourProvider` instead, I'm pretty sure.

File: osu.Game/Overlays/Profile/Header/Components/PreviousUsernames.cs [hunk 2]
```diff
@@ -84,6 +86,7 @@ public PreviousUsernames()
                                 RelativeSizeAxes = Axes.X,
                                 AutoSizeAxes = Axes.Y,
                                 Direction = FillDirection.Full,
+                                AlwaysPresent = true,
```
Comment (bdach) [inline osu.Game/Overlays/Profile/Header/Components/PreviousUsernames.cs:89]: Why? Every `AlwaysPresent` usage should ideally come with explanation. Ideally as an inline comment.

**Review by bdach — CHANGES_REQUESTED** [review]

**Comment by Joehuu** [comment]
I'd hope that this would overlap the cover (like web) to make it look better popping out

firefox_o5yMdcEBKh

**Comment by ItsShamed** [comment]
Well, the problem comes from the layout itself: it's too low, and it's even lower when the user does not have a title.

You can test adding a title to the user to see the difference:
```diff
diff --git a/osu.Game.Tests/Visual/Online/TestSceneUserProfileHeader.cs b/osu.Game.Tests/Visual/Online/TestSceneUserProfileHeader.cs
index 4f28baa849..2b32f9f653 100644
--- a/osu.Game.Tests/Visual/Online/TestSceneUserProfileHeader.cs
+++ b/osu.Game.Tests/Visual/Online/TestSceneUserProfileHeader.cs
@@ -133,7 +133,8 @@ public void TestPreviousUsernames()
                         Data = Enumerable.Range(2345, 85).ToArray()
                     },
                 },
-                PreviousUsernames = new[] { "tsrk.", "quoicoubeh", "apagnan", "epita" }
+                PreviousUsernames = new[] { "tsrk.", "quoicoubeh", "apagnan", "epita" },
+                Title = "Random title"
             }, new OsuRuleset().RulesetInfo));
         }
     }
```


image

I don't plan to deal with that in this PR.

**Comment by ItsShamed** [comment]
I could make it offset upwards a little bit like in web, but that doesn't look right imo. Unless stated otherwise.

**Review by peppy — APPROVED** [review]
Probably fine.

**Pull request merged**